src/commands/authorizeAuthorization.ts

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { URLSearchParams } from 'url';
+import * as vscode from 'vscode';
+import { IActionContext } from "vscode-azureextensionui";
+import { ApimService } from '../azure/apim/ApimService';
+import { ILoginLinkRequestContract } from '../azure/apim/contracts';
+import { AuthorizationTreeItem } from "../explorer/AuthorizationTreeItem";
+import { ext } from "../extensionVariables";
+import { localize } from '../localize';
+
+export const authorizeAuthorizationCallbackPath = "authorize-authorization-callback";
+
+export async function authorizeAuthorization(context: IActionContext, node?: AuthorizationTreeItem): Promise<void> {
+    if (!node) {
+        const authorizationNode = <AuthorizationTreeItem>await ext.tree.showTreeItemPicker(AuthorizationTreeItem.contextValue, context);
+        node = authorizationNode;
+    }
+
+    const extensionId = "ms-azuretools.vscode-apimanagement";
+    const postLoginRedirectUrl = `vscode://${extensionId}/${authorizeAuthorizationCallbackPath}`;
+
+    const apimService = new ApimService(node.root.credentials, node.root.environment.resourceManagerEndpointUrl, node.root.subscriptionId, node.root.resourceGroupName, node.root.serviceName);
+    
+    const requestBody: ILoginLinkRequestContract = {
+        postLoginRedirectUrl: postLoginRedirectUrl
+    };
+
```

Comment:
you can store the value with ext.context.globalState and retrieve it after authorization from this key value store as part of the callback handler